MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockMmReportStatusCodeHandler.h

```diff
@@ -0,0 +1,50 @@
+/** @file MockMmReportStatusCodeHandler.h
+  This file declares a mock of Standalone MM Report Status Code Handler Protocol.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_MM_REPORT_STATUS_CODE_HANDLER_PROTOCOL_H
+#define MOCK_MM_REPORT_STATUS_CODE_HANDLER_PROTOCOL_H
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Library/ReportStatusCodeLib.h>
+  #include <Protocol/MmReportStatusCodeHandler.h>
+}
+
+struct MockReportStatusCodeHandler {
+  MOCK_INTERFACE_DECLARATION (MockReportStatusCodeHandler);
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    Register,
+    (
+     IN EFI_MM_RSC_HANDLER_CALLBACK   Callback)
+    );
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    Unregister,
+    (IN EFI_MM_RSC_HANDLER_CALLBACK Callback)
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockReportStatusCodeHandler);
+MOCK_FUNCTION_DEFINITION (MockReportStatusCodeHandler, Register, 1, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockReportStatusCodeHandler, Unregister, 1, EFIAPI);
+
+EFI_MM_RSC_HANDLER_PROTOCOL  MmRscHandlerProtocol = {

```

Comment:
Based on the latest discussions in Edk2, the definition of MmRscHandlerProtocol should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9

Reply:
Change updated